File: CS/ThemeSelector/SmallControls.cs
using System;
using System.Windows;
using System.Windows.Controls.Primitives;
using DevExpress.Xpf.Core;
using ThemeSelector.Layout;
using System.Windows.Controls;

namespace ThemeSelector {
    public class ThemeButtonControl : Control {
        #region Dependency Properties
        public static readonly DependencyProperty ThemeProperty;
        public static readonly DependencyProperty ThemeSelectorLayoutControlProperty;
        static ThemeButtonControl() {
            Type ownerType = typeof(ThemeButtonControl);
            ThemeProperty = DependencyProperty.Register("Theme", typeof(Theme), ownerType, new PropertyMetadata(null));
            ThemeSelectorLayoutControlProperty = DependencyProperty.Register("ThemeSelectorLayoutControl", typeof(ThemeSelectorLayoutControl), ownerType, new PropertyMetadata(null));
        }
        #endregion
        ButtonBase button;

        public ThemeButtonControl() {
            DefaultStyleKey = typeof(ThemeButtonControl);
        }
        public Theme Theme { get { return (Theme)GetValue(ThemeProperty); } set { SetValue(ThemeProperty, value); } }
        public ThemeSelectorLayoutControl ThemeSelectorLayoutControl { get { return (ThemeSelectorLayoutControl)GetValue(ThemeSelectorLayoutControlProperty); } set { SetValue(ThemeSelectorLayoutControlProperty, value); } }
        void OnButtonClick(object sender, RoutedEventArgs e) {
            Theme theme = ((ButtonBase)sender).DataContext as Theme;
            ThemeSelectorLayoutControl.SelectValue(theme, false);
        }
        public override void OnApplyTemplate() {
            base.OnApplyTemplate();
            button = (ButtonBase)GetTemplateChild("Button");
            button.Click += OnButtonClick;
        }
    }
    public class CloseButtonControl : Control {
        ButtonBase button;

        public CloseButtonControl() {
            DefaultStyleKey = typeof(CloseButtonControl);
        }
        protected virtual void OnButtonClick(object sender, RoutedEventArgs e) { }
        public override void OnApplyTemplate() {
            base.OnApplyTemplate();
            button = (ButtonBase)GetTemplateChild("Button");
            button.Click += OnButtonClick;
        }
    }
    public class CloseThemeSelectorButtonControl : CloseButtonControl {
        protected override void OnButtonClick(object sender, RoutedEventArgs e) {
            base.OnButtonClick(sender, e);
            // close
        }
    }
}

File: VB/ThemeSelector/SmallControls.vb
' Developer Express Code Central Example:
' How to use the Theme Selector that is used in DemoCenter
' 
' This sample illustrates how to use our theme selector that is used in our demos
' 
' You can find sample updates and versions for different programming languages here:
' http://www.devexpress.com/example=E3392

Imports System
Imports System.Windows
Imports System.Windows.Controls.Primitives
Imports DevExpress.Xpf.Core
Imports ThemeSelector.Layout
Imports System.Windows.Controls

Namespace ThemeSelector
    Public Class ThemeButtonControl
        Inherits Control

        #Region "Dependency Properties"
        Public Shared ReadOnly ThemeProperty As DependencyProperty
        Public Shared ReadOnly ThemeSelectorLayoutControlProperty As DependencyProperty
        Shared Sub New()
            Dim ownerType As Type = GetType(ThemeButtonControl)
            ThemeProperty = DependencyProperty.Register("Theme", GetType(Theme), ownerType, New PropertyMetadata(Nothing))
            ThemeSelectorLayoutControlProperty = DependencyProperty.Register("ThemeSelectorLayoutControl", GetType(ThemeSelectorLayoutControl), ownerType, New PropertyMetadata(Nothing))
        End Sub
        #End Region
        Private button As ButtonBase

        Public Sub New()
            DefaultStyleKey = GetType(ThemeButtonControl)
        End Sub
        Public Property Theme() As Theme
            Get
                Return DirectCast(GetValue(ThemeProperty), Theme)
            End Get
            Set(ByVal value As Theme)
                SetValue(ThemeProperty, value)
            End Set
        End Property
        Public Property ThemeSelectorLayoutControl() As ThemeSelectorLayoutControl
            Get
                Return DirectCast(GetValue(ThemeSelectorLayoutControlProperty), ThemeSelectorLayoutControl)
            End Get
            Set(ByVal value As ThemeSelectorLayoutControl)
                SetValue(ThemeSelectorLayoutControlProperty, value)
            End Set
        End Property
        Private Sub OnButtonClick(ByVal sender As Object, ByVal e As RoutedEventArgs)

            Dim theme_Renamed As Theme = TryCast(DirectCast(sender, ButtonBase).DataContext, Theme)
            ThemeSelectorLayoutControl.SelectValue(theme_Renamed, False)
        End Sub
        Public Overrides Sub OnApplyTemplate()
            MyBase.OnApplyTemplate()
            button = CType(GetTemplateChild("Button"), ButtonBase)
            AddHandler button.Click, AddressOf OnButtonClick
        End Sub
    End Class
    Public Class CloseButtonControl
        Inherits Control

        Private button As ButtonBase

        Public Sub New()
            DefaultStyleKey = GetType(CloseButtonControl)
        End Sub
        Protected Overridable Sub OnButtonClick(ByVal sender As Object, ByVal e As RoutedEventArgs)
        End Sub
        Public Overrides Sub OnApplyTemplate()
            MyBase.OnApplyTemplate()
            button = CType(GetTemplateChild("Button"), ButtonBase)
            AddHandler button.Click, AddressOf OnButtonClick
        End Sub
    End Class
    Public Class CloseThemeSelectorButtonControl
        Inherits CloseButtonControl

        Protected Overrides Sub OnButtonClick(ByVal sender As Object, ByVal e As RoutedEventArgs)
            MyBase.OnButtonClick(sender, e)
            ' close
        End Sub
    End Class
End Namespace
